### ISSUE TITLE
[Struct2depth] Bad Motion inference on custom data

### ISSUE THREAD

**Issue by anakita** [tensorflow/models#8512]
Nice work @aneliaangelova and @VincentCa and team !
I trained the struct2depth model for motion and depth estimation for 100 epochs.The training data comprises of 1600 left and 1600 right images comprising of vehicles on indian roads.The egomotion inference values show that the vehicle moves sideways and back and front continuously , as opposed to moving in one direction.
I am not sure why am I getting such results.
Any help is appreciated.

**Comment by anakita** [comment]
@aneliaangelova or @ @VincentCa is this happening due to less number of images? Or is it because there are too many masks in a single image as in too many objects in a image?

**Comment by aneliaangelova** [comment]
Hi, 
This code has been used by many users successfully to train and test, it doesn't seem like this is a bug with the struct2depth code, but something data specific.
